File: app/src/main/java/com/recursia/popularmovies/presentation/views/SearchScreenContract.java
package com.recursia.popularmovies.presentation.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.recursia.popularmovies.domain.models.Movie;

import java.util.List;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface SearchScreenContract extends MvpView {

    void setMovies(List<Movie> movies);

    void addMovies(List<Movie> movies);

    @StateStrategyType(SkipStrategy.class)
    void openDetailScreen(Movie movie);

    @StateStrategyType(SkipStrategy.class)
    void showErrorMessage(String message);

}

File: app/src/main/java/com/recursia/popularmovies/presentation/views/contracts/SearchScreenContract.kt
package com.recursia.popularmovies.presentation.views.contracts

import com.recursia.popularmovies.domain.models.Movie
import moxy.MvpView
import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.viewstate.strategy.SkipStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(AddToEndSingleStrategy::class)
interface SearchScreenContract : MvpView {

    fun setMovies(movies: List<Movie>)

    fun addMovies(movies: List<Movie>)

    @StateStrategyType(SkipStrategy::class)
    fun showErrorMessage(message: String)
}
